### ISSUE TITLE
`continue` incorrectly marked as uncovered on python 3.9

### ISSUE THREAD

**Issue by jakkdl** [coveragepy/coveragepy#1794]
**Describe the bug**
This one random line is being marked as uncovered on python3.9, despite it clearly being executed.

**To Reproduce**
How can we reproduce the problem?  Please *be specific*. Don't link to a failing CI job. Answer the questions below:
1. What version of Python are you using?
3.9.19
The problem does *not* appear on py310, 311, 312 or 313.

1. What version of coverage.py shows the problem? The output of `coverage debug sys` is helpful.
7.5.3, I also tried 7.4.4 and 7.4.0 with the same behaviour
1. What versions of what packages do you have installed? The output of `pip freeze` is helpful.
```
attrs==23.2.0
coverage==7.5.3
exceptiongroup==1.2.1
flake8==7.0.0
flake8-async @ file:///tmp/repro_coverage_fail/flake8-async
hypothesis==6.103.0
hypothesmith==0.3.3
iniconfig==2.0.0
lark==1.1.9
libcst==1.4.0
mccabe==0.7.0
packaging==24.0
pluggy==1.5.0
pycodestyle==2.11.1
pyflakes==3.2.0
pytest==8.2.1
PyYAML==6.0.1
sortedcontainers==2.4.0
tomli==2.0.1
```
4. What code shows the problem?  Give us a specific commit of a specific repo that we can check out. If you've already worked around the problem, please provide a commit before that fix.
https://github.com/jakkdl/flake8-async/tree/async111_asyncio
commit https://github.com/jakkdl/flake8-async/commit/2cf2519198db2adbd52581e481acf9a054bbe294
5. What commands should we run to reproduce the problem? *Be specific*. Include everything, even `git clone`, `pip install`, and so on. Explain like we're five!

```sh
git clone https://github.com/jakkdl/flake8-async.git
cd flake8-async
git checkout 2cf2519198db2adbd52581e481acf9a054bbe294
python3.9 -m venv .venv
source .venv/bin/activate
pip install . coverage hypothesis hypothesmith pytest
coverage run -m pytest -k async112-trio-noautofix-normal
coverage html
```
Line 131 (`continue`) in the file `flake8_async/visitors/visitors.py` is marked as uncovered, despite line 130 (`assert True`) being marked as covered.

**Expected behavior**
The line should be marked as covered, as it is clearly being executed. If you replace the `assert True` on line 130 with `assert False` you get an `AssertionError` when running the test.

**Additional context**
First encountered in CI run https://github.com/python-trio/flake8-async/actions/runs/9287086418/job/25558523200?pr=264#step:5:1
PR: https://github.com/python-trio/flake8-async/pull/264

**Comment by nedbat** [comment]
This is due to how Python traced code in 3.9.  It's been fixed in 3.10.  The coverage.py issue about it in #198.

**Comment by jakkdl** [comment]
thank you, and sorry for the duplicate! I somehow didn't find the issue when searching
